paddle/fluid/pybind/ir.cc

```diff
@@ -0,0 +1,112 @@
+// Copyright (c) 2018 PaddlePaddle Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "paddle/fluid/pybind/ir.h"
+#include <string>
+#include <unordered_map>
+#include "paddle/fluid/framework/ir/graph.h"
+#include "paddle/fluid/framework/ir/node.h"
+#include "paddle/fluid/framework/op_desc.h"
+#include "paddle/fluid/framework/var_desc.h"
+#include "pybind11/stl.h"
+
+namespace py = pybind11;
+using paddle::framework::ir::Graph;
+using paddle::framework::ir::Node;
+using paddle::framework::OpDesc;
+using paddle::framework::ProgramDesc;
+using paddle::framework::VarDesc;
+using pybind11::return_value_policy;
+
+namespace paddle {
+namespace pybind {
+std::vector<Node *> GetNodesByType(const Node::Type &type, const Graph &graph);
+std::unordered_map<std::string, Node *> GetNodesMapByType(
+    const Node::Type &type, const Graph &graph);
+int Index(const Graph &graph, const Node &op);
+Node *Var(const Graph &graph, const std::string &name);
+Node *Var(const Graph &graph, const std::string &name);
+Node *CreateVar(Graph &graph, const std::string &name, const Node::Type &type,
+                const std::vector<int64_t> &shape);
+bool RemoveVar(Graph &graph, const std::string &name);
+bool RemoveOp(Graph &graph, int index);
+std::shared_ptr<Graph> Clone(Graph &graph);
```

Comment:
what is this for? Try to avoid `Clone` if possible

Reply:
fixed.